cli/tests/main_test.go

```diff
@@ -0,0 +1,80 @@
+package tests
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+// TestMain builds the rudder-cli binary once, exposes it via PATH and then
+// runs the package tests. It honours the cli/logging and cli/testing rules by
+// printing only essential information and performing cleanup after execution.
+func TestMain(m *testing.M) {
+	tmpDir, err := os.MkdirTemp("", "rudder-cli-bin-*")
+	if err != nil {
+		fmt.Println("failed to create temp dir:", err)
+		os.Exit(1)
+	}
+
+	defer os.RemoveAll(tmpDir)
+
+	exec, err := NewCmdExecutor("")
+	if err != nil {
+		fmt.Println("failed to init executor:", err)
+		os.Exit(1)
+	}
+
+	bin, err := NewCLIBinary(tmpDir, exec)
```

Comment:
Do we need tmpDir passed as argument given that NewCLIBinary can already handle that? Isn't it better for the CLIBinary being responsible for both creating temp directories/files as well as cleaning them up?

Reply:
Earlier temp directory creation was supposed to shared between binary and config files. I envisioned a flow where once I have a binary, I can have multiple locations of the config file and then provide the binary which config file location to use when running a e2e test. 

Let me think and refactor this flow a bit